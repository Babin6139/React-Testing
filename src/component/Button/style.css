button{
    width: 10vw;
    margin-top: 1.3rem;
    border-radius: 100rem;
}

button:hover{
    cursor: pointer;
    background-color: aliceblue;
}