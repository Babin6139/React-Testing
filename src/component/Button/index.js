import React,{useState} from 'react'
import "./style.css"

function Button({text}) {
    const [showText,setShowText]=useState(false);

    const handleOnClick=()=>{
        setShowText(!showText);
    }
  return (
    <div className="button">
        <button data-testid="Button_Id" onClick={handleOnClick}>
            Click me
        </button>
        {showText&&<div>{text}</div>}
    </div>
  )
}

export default Button