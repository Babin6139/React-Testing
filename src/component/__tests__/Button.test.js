import React from "react";
import {render,fireEvent} from "@testing-library/react";
import Button from "../Button";
import userEvent from "@testing-library/user-event";

describe("Tests for Button Component",()=>{
    it("snapshots button component",()=>{
        const mockText="Clicked the button";
        const {asFragment}=render(<Button text={mockText}/>);
        expect(asFragment(<Button text={mockText}/>)).toMatchSnapshot();
    });

    it("should render button component",()=>{
        const mockText="Clicked the button";
        const {getByText}=render(<Button text={mockText}/>);
        expect(getByText("Click me")).toBeInTheDocument;
    });

    it("we can also find by role",()=>{
        const mockText="Clicked the button";
        const {getByRole}=render(<Button text={mockText}/>);
        expect(getByRole("button")).toBeInTheDocument;
    });

    it("we can also find by the test id",()=>{
        const mockText="Clicked the button";
        const {getByTestId}=render(<Button text={mockText}/>);
        expect(getByTestId("Button_Id")).toBeInTheDocument;
    });

    it("does not show text when button is not clicked",()=>{
        const mockText="Clicked the button";
        const {queryByText}=render(<Button text={mockText}/>);
        //getByText throws an error so we use queryByText which throws null
        expect(queryByText(mockText)).toBeNull();
    });

    it("shows text when it is clicked",async ()=>{
        const mockText="Clicked the button";
        const {getByText,getByRole}=render(<Button text={mockText}/>);

        const myButton=getByRole("button");
        await userEvent.click(myButton);
        expect(getByText(mockText)).toBeInTheDocument;
    });

    it("does not show the text when clicked twice",()=>{
        const mockText="Clicked the button";
        const {queryByText,getByRole}=render(<Button text={mockText}/>);

        const myButton=getByRole("button");
        fireEvent.click(myButton);
        fireEvent.click(myButton);
        expect(queryByText(mockText)).not.toBeInTheDocument;
    })
})