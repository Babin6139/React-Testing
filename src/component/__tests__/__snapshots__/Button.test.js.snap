// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Button Component snapshots button component 1`] = `
<DocumentFragment>
  <div
    class="button"
  >
    <button
      data-testid="Button_Id"
    >
      Click me
    </button>
  </div>
</DocumentFragment>
`;
